lib/livebook/k8s/auth.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Livebook.K8s.Auth do
+  # Implementation of Access Review checks for the authenticated user using the
+  # [`SelfSubjectAccessReview`](https://kubernetes.io/docs/reference/kubernetes-api/authorization-resources/self-subject-access-review-v1/#SelfSubjectAccessReviewSpec)
+  # resource
+
+  @doc """
+  Concurrently reviews access according to a list of `resource_attributes`.
+  Expects `req` to be prepared for `SelfSubjectAccessReview`.
+  """
+  @spec batch_check(Req.Request.t(), [keyword()]) ::
+          [:ok | {:error, %Req.Response{}} | {:error, Exception.t()}]
+  def batch_check(req, resource_attribute_list) do
+    resource_attribute_list
+    |> Enum.map(&Task.async(fn -> can_i?(req, &1) end))
+    # 30 second timeout
+    |> Task.await_many(30_000)
```

Comment:
Each request itself has a timeout, so we expect all the tasks to finish eventually, so no need for extra timeout :)

```suggestion
    |> Task.await_many(:infinity)
```

Reply:
yup right.